Update library dependencies

MatiasTiainen opened the pull request:
# What
In this PR, we update library dependencies to support pandas >=2.0.0 versions to be used with this library

# Why
It is best practice to have up-to-date dependencies since currently, the library does not support pandas >=2.0.0 and thus requires other projects to have pandas <2.0.0

# Good-to-know
Successfully managed to update dependencies with this fork that previously conflicted. Related to this Issue: https://github.com/fourkind/daffy/issues which I seem to be unable to link to this PR

MatiasTiainen committed:
Update library dependencies

Daffy 0.6.0 conflicts currently with pandas >=2.0.0 versions since it depends on pandas (>=1.5.1,<2.0.0)

MatiasTiainen commented:
Referred to the changes in own pyproject.toml: 
```toml
daffy = { git = "https://github.com/MatiasTiainen/daffy.git", branch = "chore/support-pandas-2.0.0-and-newer" }
```

And the run successfully:

```zsh
poetry self add poetry-plugin-up
poetry up --latest
```

which resulted in our project to update pandas from 1.5.3 -> 2.1.3 successfully without conflicts as previously

MatiasTiainen commented:
Do you @laurimak have permission to approve the workflow and then merge this branch to the master?

laurimak commented:
It seems I don't have the permission. @H-J-Kasurinen, could please give a hand :)

MatiasTiainen committed:
Cover python version 3.11 in GHA workflow and remove 3.7

MatiasTiainen committed:
Update Github actions version in main.yml

MatiasTiainen committed:
Add .idea/ to .gitignore

MatiasTiainen committed:
Update .pre-commit-config.yaml accordingly

MatiasTiainen commented:
GitHub Actions main.yml workflow failed since `snok` does not support Python 3.7 anymore: https://github.com/snok/install-poetry/commit/b6d245985800304d1e43d05982e4b070e3458066. Fixed here in this commit: [5074c8d](https://github.com/fourkind/daffy/pull/7/commits/5074c8df017009315146f0ec14b8d28b86787707)

I also included minor fine-tuning where I updated GitHub Actions in main.yml, and aligned `.pre-commit-config.yaml` with `pyproject.toml`

tomasheiskanen commented:
Thanks @MatiasTiainen ! 

Closes #6
